src/components/FunctionSeries.tsx

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+
+import { usePlotContext } from '../contexts/plotContext';
+import { SeriesPoint } from '../types';
+import { toNumber } from '../utils';
+
+import { LineSeries, LineSeriesProps } from './LineSeries';
+
+export interface FunctionSeriesProps extends Omit<LineSeriesProps, 'data'> {
+  getY: (x: number) => number;
+  step?: number;
+}
+
+export function FunctionSeries(props: FunctionSeriesProps) {
+  const { getY, step = 0.1, ...otherProps } = props;
+  const {
+    axisContext: { x },
```

Comment:
this should use the `xAxis` prop (default to "x").

Reply:
fixed